lib/apiservers/engine/backends/container.go

```diff
@@ -671,7 +671,11 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	} else {
 		state, err := c.containerProxy.State(vc)
 		if err != nil {
-			return err
+			if IsNotFoundError(err) {
+				cache.ContainerCache().DeleteContainer(id)
```

Comment:
FWIW (yes I'm verbose tonight) if you feel strongly about it...I'll close this PR and wrap it into the `autoremove` PR -- that's where became aware of the issue...